pkg/compose/convergence.go

```diff
@@ -476,6 +476,60 @@ func (s *composeService) createMobyContainer(ctx context.Context, project *types
 	return created, err
 }
 
+// getLinks mimics V1 compose/service.py::Service::_get_links()
+func (s composeService) getLinks(ctx context.Context, projectName string, service types.ServiceConfig, number int) []string {
+	var links []string
+	format := func(k, v string) string {
+		return fmt.Sprintf("%s:%s", k, v)
+	}
+	for _, rawLink := range service.Links {
+		linkSplit := strings.Split(rawLink, ":")
+		linkServiceName := linkSplit[0]
+		l := linkServiceName
+		if len(linkSplit) == 2 {
+			l = linkSplit[1] // linkName if informed like in: "serviceName:linkName"
+		}
+		cnts, err := s.getContainers(ctx, projectName, oneOffExclude, true, linkServiceName)
+		if err != nil {
+			return nil
+		}
+		for _, c := range cnts {
+			containerName := strings.TrimPrefix(c.Names[0], "/")
+			links = append(links,
+				format(containerName, l),
+				format(containerName, strings.Join([]string{linkServiceName, strconv.Itoa(number)}, Separator)),
+				format(containerName, strings.Join([]string{projectName, linkServiceName, strconv.Itoa(number)}, Separator)),
+			)
+		}
+	}
+
+	if service.Labels[api.OneoffLabel] == "True" {
+		cnts, err := s.getContainers(ctx, projectName, oneOffExclude, true, service.Name)
```

Comment:
please refactor into a func to avoid code duplication

Reply:
Done with a local function